### DISCUSSION TITLE
Using PDM on global environment

### DISCUSSION THREAD

**Discussion by berislavlopac** [pdm-project/pdm#1655] [Q&A] [answered]
Is it possible to use PDM without any kind of isolation (either PEP 582 or venv), directly on the system-installed Python? It could be useful when installing projects inside Docker containers, or for CI/CD when using a custom image.

**Comment by glatterf42** [comment]
There is a section on [Managing a Global Project](https://pdm-project.org/latest/usage/dependency/#manage-global-project) in the docs, does that answer your question? 
The idea seems to be that you can use the `-g/--global` flags with most commands, which will add a `pyproject.toml` for your system-installed Python, which pdm can in turn manage as usual.

**Comment by berislavlopac** [comment]
Hm that is not quite related to what I had in mind. My question was about using `pdm install` to deploy a project directly on a server -- usually a container, which is in itself a form of isolation and does not require an additional level. My goal was to use `pyproject.toml` but not `pdm.lock` to build a deployment artifact (in this case a docker image).

I have ended up using a different workflow, with a custom PDM script that exports the dependencies into a `requirements.txt` file, along the lines of:

```pdm export --prod --without-hashes --format requirements --output requirements.txt```

Then, in the Dockerfile, I install `pip-tools` and run `pip-sync` against that requirements.file.

**Reply by frostming** [reply, marked as answer]
The document is related but you don't read it right. This does the trick:
```
pdm install --global --project .
```
Or the short form:
```
pdm install -gp .
```

**Comment by berislavlopac** [comment]
Thank you @frostming, that pretty much answers my original question.
